src/updatechecker.cpp

```diff
@@ -254,32 +289,65 @@ void UpdateChecker::PerformUpdateCheck(bool manual)
             }
         }
 
+        const std::string currentVersion = WideToAnsi(Settings::GetAppBuildVersion());
+
+        Settings::WriteConfigValue("LastCheckTime", time(NULL));
+
         if (!appcast.ReleaseNotesURL.empty())
             CheckForInsecureURL(appcast.ReleaseNotesURL, "release notes");
         if (!appcast.DownloadURL.empty())
             CheckForInsecureURL(appcast.DownloadURL, "update file");
 
-        Settings::WriteConfigValue("LastCheckTime", time(NULL));
+        if (appcast.SilentInstall)
+        {
+            // Check if our version is out of date.
+            if (!appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0)
+            {
+                // The same or newer version is already installed.
+                return;
+            }
 
-        const std::string currentVersion =
-                WideToAnsi(Settings::GetAppBuildVersion());
+            if (appcast.DownloadURL != "")
+            {
+                const std::wstring tmpdir = winsparkle::CreateUniqueTempDirectory();
+                Settings::WriteConfigValue("UpdateTempDir", tmpdir);
 
-        // Check if our version is out of date.
-        if ( !appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0 )
-        {
-            // The same or newer version is already installed.
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
+                UpdateDownloadSink sink(*this, tmpdir);
+                DownloadFile(appcast.DownloadURL, &sink, this);
+                sink.Close();
 
-        // Check if the user opted to ignore this particular version.
-        if ( ShouldSkipUpdate(appcast) )
-        {
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
+                if (Settings::HasDSAPubKeyPem())
+                {
+                    SignatureVerifier::VerifyDSASHA1SignatureValid(sink.GetFilePath(), appcast.DsaSignature);
+                }
+                else
+                {
+                    // backward compatibility - accept as is, but complain about it
+                    LogError("Using unsigned updates!");
```

Comment:
Why do we allow this? This seems like a back door to bypass checking in the future.